### ISSUE TITLE
Duplicate HTML content for Gmail

### ISSUE THREAD

**Issue by igorjos** [nodemailer/mailparser#160]
Hi I'm using the mail parser in order to parse multiple messages which is great, but I have problem with the HTML content in the messages where message is parsed twice in the same body. 

It doesn't matter if I parse one message or multiple the HTML is always duplicated. 

As for Text body the content is not duplicated but it doesn't render properly with new lines.

I have tested this on Windows 10 and on Ubuntu 14.04 with the latest version of the parser, and node 6.6.0 + node-imap lib the result is the same.

+Additionaly the content from Hotmail/Outlook is always returned back as text even when HTML is present in the message, but this can be treated as separate issue.

Thanks.

**Comment by andris9** [comment]
Could you provide a minimal test case example script to verify this? I can't do much with the description only

**Comment by igorjos** [comment]
Here is a part of the code that actually takes care about message parsing:

`sequencer.on('message', function(msg, seqno) 
        {  
            var parser = new MailParser();
            var body = '';
            
            clearTimeout(timer);
            
            msg.on("body", function(stream, info)
            {
                var buffer = '';
                
                stream.setEncoding('utf8');
                
                stream.on('data', function(chunk) 
                {
                    //buffer += chunk.toString('utf8');     
                    body += chunk;
                });
                
                stream.once('end', function() 
                {
                   
                    parser.write(body.toString('UTF-8'));                                            
                });
                
            });
                                   
            msg.once("end", function()
            {   
                console.log('MESSAGE LOAD ENDS');
                parser.end();	
                timer = setTimeout(function(){ parseIsDone() }, 2000);
            });
            
            parser.on('end', function(mailObj)
            {
                console.log('PARSER ENDED PARSING');
                messages.push(mailObj);
                
            });
            
        });
        
        sequencer.once('end', function()
        {
            console.log('SEQUENCER IS DONE, WAITING ON PARSER');
            
        });
        
        function parseIsDone()
        {
            console.log('PARSER DONE EVENT');
            def.resolve(messages);
            messages = [];
            clearTimeout(timer);
        };`

As you can see I'm using the sequencer from node-imap: 
`var f = localImap.seq.fetch(from+':'+to, { bodies: ['HEADER.FIELDS (FROM SUBJECT)',''] });`
where localimap is the IMAP connection to GMAIL 
once it fetch the messages it calls for the method from the above and parse each message by using promises. 

As addition to my first comment not all messages are having duplicate HTML, as I can see PayPal messages are okay, but Aliexpress for example are duplicated in the content.

In case you need more info please let me know, meanwhile I will do some workaround like in case content is duplicated to replace it or smth.

Thanks

**Comment by andris9** [comment]
Can you provide such example message as well that has duplicate contents?

**Comment by andris9** [comment]
You should check whats exactly the data that you write to Mailparser. Are you sure that you do not get the html content twice from the message 'body' event?

**Comment by igorjos** [comment]
You need (nodejs) message object or to send you that message to your mail? (Both fits for me)
